crates/air-lib/interpreter-cid/src/lib.rs

```diff
@@ -121,7 +123,26 @@ impl std::error::Error for CidCalculationError {
 }
 
 /// Calculate a CID of JSON-serialized value.
-pub fn value_to_json_cid<Val: Serialize>(value: &Val) -> Result<CID<Val>, CidCalculationError> {
-    let data = serde_json::to_vec(value)?;
-    Ok(json_data_cid(&data))
+// TODO we might refactor this to `SerializationFormat` trait
+// that both transform data to binary/text form (be it JSON, CBOR or something else)
+// and produces CID too
+pub fn value_to_json_cid<Val: Serialize + ?Sized>(
+    value: &Val,
+) -> Result<CID<Val>, CidCalculationError> {
+    use cid::Cid;
+    use multihash::{Code, MultihashDigest};
+    use sha2::Digest;
+
+    let mut hasher = sha2::Sha256::new();
+    serde_json::to_writer(BufWriter::with_capacity(8 * 1024, &mut hasher), value)?;
```

Comment:
IMHO this 8k buffer is a bit overkill for small values. What if this method adopts using the value size that can be sent from an upper function?

Reply:
Our memory footprint is dozens of megabytes.  Allocating 8kb takes same time as allocating 512b.